peloton/update_project.py

```diff
@@ -0,0 +1,1371 @@
+    ) -> list[dict]:
+        return [
+            # Expecting a DataFrame where the required column is the index.
+            (self._inputs_default | dict(content_id=row.name))
+            for _, row in data_frame.iterrows()
+        ]
+
+    def _add_selector(
+        self,
+        mutation: sgqlc.operation.Operation,
+        index: int
+    ) -> None:
+        add = mutation.add_project_v2_item_by_id(
+            input=self._inputs[index], __alias__=self._aliases[index]
+        )
+        add.item().id()
+
+    def _get_data_from_sub_mutations(
+        self,
+        payloads: list[list[github_schema.AddProjectV2ItemByIdPayload]]
+    ) -> list[str]:
+        result = []
+        for sub_list in payloads:
+            ids_in_project = [p.item.id for p in sub_list]
+            result.extend(ids_in_project)
+        return result
+
+
+class AddDraftsMutation(PaginatedMutation):
+    """
+    Mutation to add Discussions to the Peloton GitHub project.
+
+    Added by adding project 'drafts' - via a given title and body text - and
+    records (self.results) the IDs of the corresponding items in the project.
+    """
+    def _make_inputs(
+        self,
+        data_frame: pd.DataFrame,
+        project_cols: ProjectItemsQuery.ColNames,
+        issue_cols: IssuesQuery.ColNames,
+    ) -> list[dict]:
+        inputs = []
+        for _, row in data_frame.iterrows():
+            title = row[issue_cols.title]
+            # TODO: how to avoid bad unicode escape sequence with emojis?
+            #  sgqlc uses bytes(op).decode('utf-8') on its queries which
+            #  might be the
+            #  source of the problem.
+            title = "".join(filter(lambda t: t in set(printable), title))
+            inputs.append((
+                self._inputs_default
+                |
+                dict(title=title, body=row[issue_cols.url])
+            ))
+        return inputs
+
+    def _add_selector(
+        self,
+        mutation: sgqlc.operation.Operation,
+        index: int
+    ) -> None:
+        add = mutation.add_project_v2_draft_issue(
+            input=self._inputs[index], __alias__=self._aliases[index]
+        )
+        add.project_item().id()
+
+    def _get_data_from_sub_mutations(
+        self,
+        payloads: list[list[github_schema.AddProjectV2DraftIssuePayload]]
+    ) -> list[str]:
+        result = []
+        for sub_list in payloads:
+            ids_in_project = [p.project_item.id for p in sub_list]
+            result.extend(ids_in_project)
+        return result
+
+
+class ClearPelotonDateMutation(PaginatedMutation):
+    """
+    Mutation clearing next-peloton-date field from items in the Peloton project.
+
+    Used when there has been an update to the issue/discussion in question,
+    meriting that it be revisited in an upcoming Peloton meeting.
+    """
+    _inputs_default = (
+        PaginatedMutation._inputs_default
+        |
+        dict(field_id="PVTF_lADOABU7f84ALhAIzgP3vFs")
```

Comment:
@trexfeathers eek!

Could this be a class member constant with some explanation as to what it is?

Reply:
65ed1b36807721b629547d9115ddc7f4f2b1e124